Neos.Flow/Classes/Http/Helper/UploadedFilesHelper.php

```diff
@@ -21,17 +23,26 @@
 abstract class UploadedFilesHelper
 {
     /**
-     * @param UploadedFileInterface[] $uploadedFiles
+     * @param UploadedFileInterface[]|mixed[][] $uploadedFiles A (deep) array of UploadedFile or an untangled $_FILES array
      * @param array $arguments
      * @param array $currentPath internal argument for recursion
-     * @return array The nested array of paths and uploaded files
+     * @return UploadedFileInterface[] The nested array of paths and uploaded files
```

Comment:
That's not technically 100% correct but... it's more specific than just `array`?

Reply:
psalm is with me:
```
ERROR: InvalidReturnType - Packages/Framework/Neos.Flow/Classes/Http/Helper/UploadedFilesHelper.php:29:16 - The declared return type 'array<array-key, Psr\Http\Message\UploadedFileInterface>' for Neos\Flow\Http\Helper\UploadedFilesHelper::upcastUploadedFiles is incorrect, got 'array<array-key, Neos\Http\Factories\FlowUploadedFile|array<array-key, Psr\Http\Message\UploadedFileInterface>>'
     * @return UploadedFileInterface[] The nested array of paths and uploaded files
```

Reply:
Not sure how this is done correctly in psalm... https://psalm.dev/r/cc875a4a60
Maybe we need to stick with the generic `array` 🤷‍♂

Reply:
Well, not possible with psalm as of now (and in the foreseeable future):
https://github.com/vimeo/psalm/issues/2777#issuecomment-583891417